scripts/actions/spells/white/sacrifice.lua

```diff
@@ -10,25 +11,48 @@ end
 spellObject.onSpellCast = function(caster, target, spell)
     local removables = { xi.effect.FLASH, xi.effect.BLINDNESS, xi.effect.PARALYSIS, xi.effect.POISON, xi.effect.CURSE_I, xi.effect.CURSE_II, xi.effect.DISEASE, xi.effect.PLAGUE }
 
-    -- remove one effect and add it to me
-    for i, effect in ipairs(removables) do
-        if target:hasStatusEffect(effect) then
-            spell:setMsg(xi.msg.basic.MAGIC_ABSORB_AILMENT)
+    -- Maximum number of effects to transfer
+    local maxTransferCount = caster:hasStatusEffect(xi.effect.AFFLATUS_SOLACE) and 7 or 1
+
+    -- Use the helper function to transfer status effects
+    local transferCount = 0  -- Initialize transferCount
 
-            local statusEffect = target:getStatusEffect(effect)
+    -- Loop over the original effects
+    for _, effect in ipairs(removables) do
+        if transferCount < maxTransferCount then
+            if target:hasStatusEffect(effect) then
+                local statusEffect = target:getStatusEffect(effect)
 
-            -- only add it to me if I don't have it
-            if not caster:hasStatusEffect(effect) then
-                caster:addStatusEffect(effect, statusEffect:getPower(), statusEffect:getTickCount(), statusEffect:getDuration())
+                -- only add it to me if I don't have it
+                if not caster:hasStatusEffect(effect) then
+                    caster:addStatusEffect(effect, statusEffect:getPower(), statusEffect:getTickCount(), statusEffect:getDuration())
+                    transferCount = transferCount + 1
+                end
+
+                target:delStatusEffect(effect)
             end
+        else
+            break  -- Break the loop if the maximum number of effects are transferred
+        end
+    end
```

Comment:
Instead of wrapping the logic inside the counter logic, we could move it to the end. Less deepness = easier to read.
```lua
    -- Maximum number of effects to transfer
    local maxTransferCount = caster:hasStatusEffect(xi.effect.AFFLATUS_SOLACE) and 7 or 1
    local transferCount    = 0  -- Initialize transferCount

    -- Loop over the original effects
    for _, effect in ipairs(removables) do
        if target:hasStatusEffect(effect) then
            local statusEffect = target:getStatusEffect(effect)

            -- only add it to me if I don't have it
            if not caster:hasStatusEffect(effect) then
                caster:addStatusEffect(effect, statusEffect:getPower(), statusEffect:getTickCount(), statusEffect:getDuration())
                transferCount = transferCount + 1
            end

            target:delStatusEffect(effect)
        end

        if transferCount >= maxTransferCount then
            break
        end
    end

Reply:
I can make some edits to this, let me try and get back to you. Thank you for the suggestion.

Reply:
Note, I'm not exactly a programmer, just a bored sysadmin who wants to contribute to fixing my favorite game! So I appreciate the patience while I dust off my programming brain